Client/Frontend/Browser/LoginsHelper.swift

```diff
@@ -210,8 +210,25 @@ class LoginsHelper: TabContentScript {
 
             let json = JSON(jsonObj)
             let src = "window.__firefox__.logins.inject(\(json.stringValue()!))"
-            self.tab?.webView?.evaluateJavaScript(src, completionHandler: { (obj, err) -> Void in
+
+            guard !json["logins"].isEmpty else {
+                self.tab?.webView?.evaluateJavaScript(src, completionHandler: { (obj, err) -> Void in
+                })
```

Comment:
Sorry for this being so late 🙂 Is there a reason you evaluate the JS here if logins isn't empty? Shouldn't it just return?